autofit/mapper/variable.py

```diff
@@ -86,3 +88,55 @@ def ndim(self) -> int:
         How many dimensions does this variable have?
         """
         return len(self.plates)
+
+
+def broadcast_plates(
+        value: np.ndarray, 
+        in_plates: Tuple[Plate, ...], 
+        out_plates: Tuple[Plate, ...], 
+        reducer: np.ufunc = np.sum
+    ) -> np.ndarray:
+    """
+    Extract the indices of a collection of plates then match
+    the shape of the data to that shape.
+
+    Parameters
+    ----------
+    value
+        A value to broadcast
+    in_plates
+        Plates representing the dimensions of the values
+    out_plates
+        Plates representing the output dimensions
+    reducer
+        function to reduce excess plates over, default np.sum
+        must take axis as keyword argument
+
+
+    Returns
+    -------
+    The value reshaped to match the plates
+    """
+    n_in = len(in_plates)
+    n_out = len(out_plates)
+    shift = np.ndim(value) - n_in
+    if shift > 1 or shift < 0:

```

Comment:
if not (0 <= shift <= 1) might be clearer

Reply:
yeah